src/document/DocumentAnnotator.ts

```diff
@@ -60,14 +60,21 @@ export default class DocumentAnnotator extends BaseAnnotator {
         const pageReferenceEl = this.getPageReference(pageEl);
         const managers = this.managers.get(pageNumber) || new Set();
         const resinTags = { fileid: fileId, iscurrent: isCurrentFileVersion };
+        let destroyManagers = false;
 
-        // Destroy any managers that were attached to page elements that no longer exist
+        // If a referenced page element doesn't exist, destroy all managers to keep them in sync
         managers.forEach(manager => {
             if (!manager.exists(pageEl)) {
-                manager.destroy();
-                managers.delete(manager);
+                destroyManagers = true;
             }
         });
+        if (destroyManagers) {
+            managers.forEach(manager => {
+                manager.destroy();
+            });
+            managers.clear();
+            destroyManagers = false;
```

Comment:
What happens if we don't reset the destroyManagers values here?

Reply:
Should be the same result, but I don't see any downsides to resetting the value since it should not be true anymore.

Reply:
Yea it should have no effects but it's not needed though because the value won't be used again after that. Resetting it makes it seem like that value is relevant even after we clear the managers.